lib/puppet/type/dsc_opticaldiskdriveletter.rb

```diff
@@ -0,0 +1,120 @@
+require 'pathname'
+
+Puppet::Type.newtype(:dsc_opticaldiskdriveletter) do
+  require Pathname.new(__FILE__).dirname + '../../' + 'puppet/type/base_dsc'
+  require Pathname.new(__FILE__).dirname + '../../puppet_x/puppetlabs/dsc_type_helpers'
+
+
+  @doc = %q{
+    The DSC OpticalDiskDriveLetter resource type.
+    Automatically generated from
+    'StorageDsc/DSCResources/MSFT_OpticalDiskDriveLetter/MSFT_OpticalDiskDriveLetter.schema.mof'
+
+    To learn more about PowerShell Desired State Configuration, please
+    visit https://technet.microsoft.com/en-us/library/dn249912.aspx.
+
+    For more information about built-in DSC Resources, please visit
+    https://technet.microsoft.com/en-us/library/dn249921.aspx.
+
+    For more information about xDsc Resources, please visit
```

Comment:
We decided this was either valid or not important for now.

Reply:
When this line in the template was written HQ DSC Resources didn't exist, so there wasn't a distinction.

Admittantly the language in the line is problematic without the 'Experimental' reference, as it should say something more like '...information about Microsoft DSC Resources we vendor...'.

FWIW I don't this this doc block is helpful at all, and should really be in the readme more than in each type.

In this PR, to 'fix' this would require touching **every** template file, whereas the point of this ticket is to only update the DSC Resources that have new versions. I have created MODULES-6642 to address whether to change this part of the template or not